### PR TITLE
Story 14

### PR DESCRIPTION
**Pull request by limsammy**
Related Issue: #14 

Summary: Set up items index and show page to reflect current status using an enum in the item model.

Author(s): sam

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class Item < ApplicationRecord
   has_many :orders_items
   has_many :orders, through: :orders_items
 
+  enum status: %w(active retired)
```
Comment (amhursh) [inline app/models/item.rb:9]: Cool. I think these are all we need.

File: app/views/items/index.html.erb [hunk 2]
```diff
@@ -8,7 +8,11 @@
 			<div class="item_card index_item_<%= item.id %>">
 				<img class="item-card-img" src="<%= item.image %>" alt="">
 				<h3><%= link_to item.title, item_path(item) %></h3>
-				<p><%= link_to "Add to Cart", cart_path(item_id: item.id), method: :post, class: "cart btn btn-outline-info" %></p>
+				<% if item.active? %>
+					<%= link_to "Add to Cart", cart_path(item_id: item.id), method: :post, class: "cart-button btn btn-outline-info" %>
```
Comment (amhursh) [inline app/views/items/index.html.erb:12]: I think this conditional should do us just fine.

File: spec/features/visitor/items/visitor_can_view_retired_items_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+      expect(page).to_not have_content("Add to Cart")
+      expect(page).to have_content("Item Retired")
+    end
+  end
+
+  context "visitor visits items index" do
+    scenario "visitor cannot add item to cart" do
+      item = create(:item, status: 1)
+      item_2 = create(:item, status: 0)
```
Comment (amhursh) [inline spec/features/visitor/items/visitor_can_view_retired_items_spec.rb:20]: Good to test both /items and /item

File: spec/features/visitor/visitor_can_add_items_to_cart_spec.rb [hunk 4]
```diff
@@ -17,7 +17,7 @@
         expect(current_path).to eq('/cart')
         expect(page).to have_xpath("//img[@src='#{item.image}']")
         expect(page).to have_content(item.title)
-        save_and_open_page
+        # save_and_open_page
```
Comment (randy-darsh) [inline spec/features/visitor/visitor_can_add_items_to_cart_spec.rb:20]: need to remove this

**Review by amhursh — APPROVED** [review]

File: app/models/item.rb (hunk 1 above)
Comment (JoelLindow) [reply app/models/item.rb:9]: My favorite thing about enums is how they give you options. Life is all about options.

File: app/views/items/index.html.erb (hunk 2 above)
Comment (randy-darsh) [reply app/views/items/index.html.erb:12]: i concur

**Review by JoelLindow — COMMENTED** [review]
This looks great. This is how the enum plugs in to the table. Pretty good stuff!

File: spec/features/visitor/visitor_can_add_items_to_cart_spec.rb (hunk 4 above)
Comment (JoelLindow) [reply spec/features/visitor/visitor_can_add_items_to_cart_spec.rb:20]: Randy's right. Leaving stuff like this in a commit it totally Busch League. Not saying I wouldn't have done it, but that doesn't mean it's ok.

**Comment by randy-darsh** [comment]
looks good. merging to dev

**Pull request merged**